calendar/page.js

```diff
@@ -319,6 +362,11 @@ async function configureGristSettings() {
   // When options changed in the widget configuration (reaction to perspective change)
   grist.onOptions(onGristSettingsChanged);
 
+  // To get types, we need to know the tableId. This is a way to get it.
+  grist.on('message', (e) => {
+    if (e.tableId) { colTypesFetcher.gotTableId(e.tableId); }
```

Comment:
Thanks. I think I still need to keep the structure where I check for mappings elsewhere because this event doesn't actually say what the mappings are, and I see no obvious way to get them.